usvm-ts/src/main/kotlin/org/usvm/machine/expr/TSExprResolver.kt

```diff
@@ -0,0 +1,555 @@
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsRightShiftExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsStaticCallExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsPtrCallExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsStrictEqExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsStrictNotEqExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsSubExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsTernaryExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsTypeOfExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsUnaryPlusExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsUnsignedRightShiftExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsVoidExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(expr: EtsYieldExpr): UExpr<out USort>? {
+        logger.warn { "visit(${expr::class.simpleName}) is not implemented yet" }
+        error("Not supported $expr")
+    }
+
+    override fun visit(value: EtsArrayAccess): UExpr<out USort>? {
+        logger.warn { "visit(${value::class.simpleName}) is not implemented yet" }
+        error("Not supported $value")
+    }
+
+    override fun visit(value: EtsInstanceFieldRef): UExpr<out USort>? = with(ctx) {
+        val instanceRef = resolve(value.instance)?.asExpr(addressSort) ?: return null
+        // TODO: checkNullPointer(instanceRef)
+        val fieldType = scene.projectAndSdkClasses
```

Comment:
Let's move this field type lookup somewhere else. This thing will be a reason of a huge performance issues. So, it would be nice if we can easily detect it.